src/main/java/service/impl/ReportServiceImpl.java

```diff
@@ -0,0 +1,35 @@
+package service.impl;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import model.FruitTransaction;
+import service.ReportService;
+import strategy.StrategyService;
+
+public class ReportServiceImpl implements ReportService {
+    private static final String SPLIT_SYMBOL = ",";
+    private static final String NEXT_LINE = "\n";
+
+    @Override
+    public String generateReport(List<FruitTransaction> transactions,
+                                 StrategyService strategyService) {
+        Map<String, Integer> reportDataMap = new HashMap<>();
+        StringBuilder reportResult = new StringBuilder();
+        for (FruitTransaction transaction : transactions) {
+            int storeQuantity = reportDataMap.get(transaction.getFruit()) == null ? 0
+                    : reportDataMap.get(transaction.getFruit());
+            reportDataMap.put(transaction.getFruit(),
+                    strategyService.get(transaction.getOperation())
+                            .operationHandler(transaction, storeQuantity));
```

Comment:
We should process all FruitTransactions after we parse List<FruitTransactions> from file, and store info about fruits in Storage. Than we could get data from Storage and create report